lib/salus/scanners/slither.rb

```diff
@@ -0,0 +1,94 @@
+require 'json'
+require 'salus/scanners/base'
+
+# Slither (https://github.com/crytic/slither) is a Python-based static
+# analysis tool for Solidity smart contracts
+
+module Salus::Scanners
+  class Slither < Base
+    include Salus::Formatting
+
+    def should_run?
+      # only support truffle/hardhat for now
+      return false if !@repository.package_json_present? || !has_package_config
+
+      @repository.sol_file_present? != false
+    end
+
+    def run
+      # Most Solidity projects will be Hardhat (https://hardhat.org/) or Truffle (https://trufflesuite.com/),
+      # which both use NPM to manage Solidity dependencies. We should NPM install in these cases
+      # to ensure Slither has the needed dependencies for scanning
+
+      pre_slither_commands = ['npm install', 'npm config set user 0']
```

Comment:
Is it possible to extract some parts of the code to separate methods. `def run` method becomes large. It's ok to have a number of small method instead of a long method. Martin Fowler has a great [article](https://martinfowler.com/bliki/FunctionLength.html) on this.

Reply:
Updated